File: trunk/base/doc/guides/ConsumerExample.cpp
#include "rgma/Consumer.h"
#include "rgma/QueryTypeWithInterval.h"
#include "rgma/RGMAException.h"
#include "rgma/TimeInterval.h"
#include "rgma/TimeUnit.h"
#include "rgma/TupleSet.h"

#include <string>
#include <iostream>

using glite::rgma::Consumer;
using glite::rgma::QueryTypeWithInterval;
using glite::rgma::RGMAException;
using glite::rgma::TimeInterval;
using glite::rgma::TimeUnit;
using glite::rgma::TupleSet;

int main(int argc, char* argsv[])
{

   try
   {
      TimeInterval historyPeriod(10, TimeUnit::MINUTES);
      TimeInterval timeout(5, TimeUnit::MINUTES);
      std::string select("SELECT userId, aString, aReal, "
            "anInt, RgmaTimestamp FROM "
            "default.userTable");
      Consumer c(select, QueryTypeWithInterval::C, historyPeriod, timeout);

      bool endOfResults = false;
      while (!endOfResults)
      {
         TupleSet tupleSet;
         c.pop(2000, tupleSet);
         if(tupleSet.begin() == tupleSet.end())
         {
            sleep(2);
         }
         else
         {
           
             TupleSet::const_iterator tuple = tupleSet.begin();
             while (tuple != tupleSet.end())
             {
                 std::cout << "userId=" << tuple->getString(0) << ", ";
                 std::cout << "aString=" << tuple->getString(1) << ", ";
                 std::cout << "aReal=" << tuple->getFloat(2) << ", ";
                 std::cout << "anInt=" << tuple->getInt(3) << std::endl;
                 ++tuple;
             }
             
         }
         endOfResults = tupleSet.isEndOfResults();
      }

      c.close();
   }

   catch (RGMAException e)
   {
      std::cerr << "R-GMA application error in Consumer."<< std::endl;
      std::cout << e.getMessage() << std::endl;
   }

}

File: trunk/base/doc/guides/ConsumerExample.java
import org.glite.rgma.Consumer;
import org.glite.rgma.QueryTypeWithInterval;
import org.glite.rgma.RGMAException;
import org.glite.rgma.TimeInterval;
import org.glite.rgma.TimeUnit;
import org.glite.rgma.Tuple;
import org.glite.rgma.TupleSet;

public class ConsumerExample {
  public static void main(String[] args) {

    try {
      TimeInterval historyPeriod = new TimeInterval(10, TimeUnit.MINUTES);
      TimeInterval timeout = new TimeInterval(5, TimeUnit.MINUTES);
      String select = "SELECT userId, aString, aReal, anInt, "
          + "RgmaTimestamp FROM default.userTable";
      Consumer consumer = new Consumer(select, QueryTypeWithInterval.C,
          historyPeriod, timeout);

      TupleSet ts = null;
      do {
        ts = consumer.pop(2000);
        if (ts.getData().size() == 0) {
          try {
            Thread.sleep(2 * 1000);
          } catch (InterruptedException e) {
          }
        } else {

          for (Tuple t : ts.getData()) {
            System.out.print("userId=" + t.getString(0) + ", ");
            System.out.print("aString=" + t.getString(1) + ", ");
            System.out.print("aReal=" + t.getFloat(2) + ", ");
            System.out.println("anInt=" + t.getInt(3));
          }

        }
      } while (!ts.isEndOfResults());

      consumer.close();

    } catch (RGMAException e) {
      System.err.println("ERROR: " + e.toString());
      System.exit(1);
    }

  }
}
